### PR TITLE
Fix fast forwarding for SubqueryExecutor

### PR DESCRIPTION
**Pull request by markuspf**
Use the bespoke skipRowsRange in SubqueryExecutor

### CODE REVIEW

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/10242/

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/10243/

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/10244/

**Comment by markuspf** [comment]
This fix is entirely broken and needs to be redone tomorrow.

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/10254/

**Comment by markuspf** [comment]
This now includes #11755 because that change is needed to make tests pass properly.

**Comment by markuspf** [comment]
Jenkins blue.

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,13 @@
 devel
 -----
 
+* Fix crash in execution of non-spliced subqueries if remainder of subquery is skipped
```
Comment (jsteemann) [inline CHANGELOG:4]:
```suggestion
* Fix crash in execution of non-spliced subqueries if remainder of subquery is skipped.
```

File: CHANGELOG [hunk 2]
```diff
@@ -1,6 +1,13 @@
 devel
 -----
 
+* Fix crash in execution of non-spliced subqueries if remainder of subquery is skipped.
+
+* Made modification subqueries non-passthrough. The passthrough logic only works
+  if exactly as many output rows are produced as input rows are injected.
+  If the subquery with modification is skipped however this API is violated,
+  we perform the subquery with the modification, but then discard the input if skipped)
```
Comment (jsteemann) [inline CHANGELOG:9]:
```suggestion
  we perform the subquery with the modification, but then discard the input if skipped).
```

**Pull request merged**